src/audio/buffer.c

```diff
@@ -159,7 +159,8 @@ void comp_update_buffer_consume(struct comp_buffer *buffer, uint32_t bytes)
 	/* calculate free bytes */
 	buffer->free = buffer->size - buffer->avail;
 
-	if (buffer->sink->is_dma_connected)
+	if (buffer->sink->is_dma_connected &&
+	    !buffer->source->is_dma_connected)
```

Comment:
sorry, I think then handles for write_back and invalidate in produce and consume here are chaotic, we don't need do this for consume, please do it for produce only.

I can do a quick patch for that.